### PR TITLE
py-person-class

### PR DESCRIPTION
**Pull request by muriakula**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,26 @@
...
+        new_person = Person.people[person["name"]]
+        if "wife" in person:
+            if person["wife"] in Person.people:
+                new_person.wife = Person.people[person["wife"]]
+        if "husband" in person:
+            if person["husband"] in Person.people:
+                new_person.husband = Person.people[person["husband"]]
+        result.append(new_person)
+    for person in result:
+        print(person)
```
Comment (Bohdan-Salamakha) [inline app/main.py:25]: print new_person in a previous loop

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result = []
+    for person in people:
+        name, age = person["name"], person["age"]
+        Person(name, age)
```
Comment (Bohdan-Salamakha) [inline app/main.py:14]: find result using list comprehension and use persons from it

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
+    [Person(person["name"], person["age"]) for person in people]
+    for person in people:
+        new_person = Person.people[person["name"]]
+        if "wife" in person:
+            if person["wife"] in Person.people:
+                new_person.wife = Person.people[person["wife"]]
+        if "husband" in person:
+            if person["husband"] in Person.people:
+                new_person.husband = Person.people[person["husband"]]
+    return list(Person.people.values())
```
Comment (Bohdan-Salamakha) [inline app/main.py:20]:
```suggestion
    person_list = [Person(person["name"], person["age"]) for person in people]
    for person in people:
        if person.get("husband"):
            wife = Person.people[person["name"]]
            husband = Person.people[person["husband"]]
            wife.husband = husband
            husband.wife = wife
    return person_list
```

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v2

**Review by olena-hul — APPROVED** [review]

**Comment by olena-hul** [comment]
Good job!

**Pull request closed without merging**